swap/tests/alice_and_bob_refund_using_cancel_and_refund_command.rs

```diff
@@ -61,40 +60,20 @@ async fn given_alice_and_bob_manually_refund_after_funds_locked_both_refund() {
 
         // Bob manually refunds
         bob_join_handle.abort();
-        let bob_state =
-            cli::refund(bob_swap.id, bob_swap.bitcoin_wallet, bob_swap.db, false).await??;
+        let bob_state = cli::refund(bob_swap.id, bob_swap.bitcoin_wallet, bob_swap.db).await?;
 
         ctx.assert_bob_refunded(bob_state).await;
 
-        // manually cancel ALice's swap (effectively just notice that Bob already
-        // cancelled and record that)
-        ctx.restart_alice().await;
-        let alice_swap = ctx.alice_next_swap().await;
-        assert!(matches!(
-            alice_swap.state,
-            AliceState::XmrLockTransactionSent { .. }
-        ));
-
-        asb::cancel(
-            alice_swap.swap_id,
-            alice_swap.bitcoin_wallet,
-            alice_swap.db,
-            false,
-        )
-        .await??;
-
         // manually refund ALice's swap
         ctx.restart_alice().await;
         let alice_swap = ctx.alice_next_swap().await;
-        assert!(matches!(alice_swap.state, AliceState::BtcCancelled { .. }));
```

Comment:
Is just deleting this okay? Shouldn't we still reach the same state?

Reply:
I removed `asb::cancel()` which was ensuring that Alice is in the `BtcCancelled` state. Since it has been removed Alice does not proceed past `XmrLockTransactionSent`.

Reply:
Are you saying there is no point in doing `asb::cancel` because it has already been performed by the CLI and hence it would fail whereas previously it would just gracefully do nothing?

Reply:
pretty much. It would advance the CLI state machine to the BtcCancelledState

Reply:
What is the resulting user behaviour now? If a user runs cancel but it has already been cancelled, shouldn't we transition to `BtcCancelled` as part of that?

In other words, is our approach now that these manual commands never save state to the DB unless they are successful?

Reply:
I think the saving to the db behaviour is unchanged in these manual commands?  I think the main change is that `asb::cancel()` is erroring if cancel has already been published which is wrong. `cli::cancel()` is performing this check. I should be able to reintroduce this assertions after I fix that.

Reply:
Well, it may be okay to assume that the manual commands (which effectively forward to `asb::{cancel, refund, ...}`) can fail with all sorts of errors if they don't work out. i.e. we could say:

`asb cancel` can fail if cancel is already published and it will not transition the internal state.

I think that is reasonable and may actually be the best choice. It makes those commands very predictable. If `cancel` fails because `cancel` is already published, the user can always run `punish` and it will work and transition the state to `punished`. To actually step through all the states `resume` should be used as you documented in the commit message!

Reply:
yeh ok. Ill had a info log statement so the user isn't confused and tries to resubmit cancel